dags/src/utility.py

```diff
@@ -90,14 +126,224 @@ def download_geodata(**kwargs):
         
             logger.info(f"Writing feature data to breedfides-airflow/wfs/{filename}.json")
-            with open(f'wfs/{filename}.json', 'w') as json_file:
+            output_dir_path = os.path.join(current_dir, 'wfs', f'{filename}.json')
+            with open(output_dir_path, 'w') as json_file:
                 json.dump(response, json_file)
-        
+                
+        elif 'fetch_cdc' in dag_id:
+            ftp = ftplib.FTP('opendata.dwd.de', timeout=3600)
+            
+            ftp.login() ### Login to FTP site
+            
+            remote_directory = f"/climate_environment/CDC/grids_germany/daily/Project_TRY/{ftp_dir}"
+            ftp.cwd(remote_directory)
+            
+            ftp_files = set(obj[:-3] for obj in ftp.nlst() if '_20' in obj) ### List the files in the remote FTP directory (data from year 2000)
+            net_cdf_files = set(os.path.basename(fp) for fp in glob.glob(f'{local_file_path}/*.nc')) ### List the files in the local directory
+            
+            file_exists, files_to_download = verify_file_exists(net_cdf_files, ftp_files)
+            
+            if file_exists:
+                pass
+            
+            else: 
+                ## Download each file
+                for file in files_to_download:
+                    logger.info(f"Downloading file : {file}")
+                    remote_file = io.BytesIO() ## Open remote file 
+                    ftp.retrbinary('RETR ' + file, remote_file.write)
+                    remote_file.seek(0)
+                    
+                    ## Decompress GZIP file
+                    with gzip.GzipFile(fileobj=remote_file, mode='rb') as gzipped_file:
+                        decompressed_content = gzipped_file.read() ## Read the decompressed content
+                
+                    logger.info(f"Writing netCDF data to {local_file_path}")
+                
+                    with open(f"{os.path.join(f'{local_file_path}/', os.path.splitext(file)[0])}", 'wb') as local_file:
+                        local_file.write(decompressed_content)
+                        
+            ## Close the FTP connection
+            ftp.quit()
+            
+        elif 'fetch_soil' in dag_id:
+            base_url = 'https://cloud.thuenen.de/index.php/s/6RTizJ5ZEPsHJps/'
+            login_url = base_url + 'authenticate/downloadShare'
+            file_url = base_url + 'download?path=%2F&files=BUEK_data.gpkg'
+
+            gpkg_files = set(os.path.basename(fp) for fp in glob.glob(f'{local_file_path}/*.gpkg')) ### List the files in the local directory
+            file_exists, files_to_download = verify_file_exists(gpkg_files, {'BUEK_data.gpkg'})
+      
+            if file_exists:
+                pass
+            
+            else:
+                with requests.Session() as s:
+                    response = s.get(login_url)
+                    soup = BeautifulSoup(response.content, 'html.parser')
```

Comment:
Using BeautifulSoup to scrape the site will work with the current version but is prone to errors on updates, which will render the frontend differently. Since the file will be stored locally, we do not need this whole part. Sorry for the misinformation on our behalf, which wasted your time here.